File: java/src/main/java/com/function/util/KeyVaultOperations.java
package com.function.util;

import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.security.keyvault.keys.KeyClientBuilder;
import com.azure.security.keyvault.keys.models.CreateRsaKeyOptions;
import com.azure.security.keyvault.keys.models.KeyOperation;
import com.azure.security.keyvault.keys.models.KeyVaultKey;

public class KeyVaultOperations {

    //Create RSA Key on Key Vault
    public static String CreateKey(String vaultEndpoint, String keyName)
    {
        var kvClient = new KeyClientBuilder()
                            .vaultUrl("https://"+vaultEndpoint)
                            .credential(new DefaultAzureCredentialBuilder().build())
                            .buildClient();

        var keyOptions = new CreateRsaKeyOptions(keyName)
                            .setEnabled(true)
                            .setKeyOperations(KeyOperation.WRAP_KEY, KeyOperation.UNWRAP_KEY);

        KeyVaultKey newKey = kvClient.createRsaKey(keyOptions);

        return newKey.getProperties().getVersion();
    }

    //Rotate RSA Key on Key Vault. New version is created if a key already exists.
    public static String RotateKey(String vaultEndpoint, String keyName)
    {
        return CreateKey(vaultEndpoint, keyName);
    }
}

File: dotnet/src/Util/KeyVaultOperations.cs
using Azure.Identity;
using Azure.Security.KeyVault.Keys;
using System;
using System.Threading.Tasks;

namespace MongoEncryption.Util
{
    internal static class KeyVaultOperations
    {
        //Create RSA Key on Key Vault
        public static async Task<string> CreateKeyAsync(string vaultEndpoint, string keyName)
        {
            var kvClient = new KeyClient(vaultUri: new Uri($"https://{vaultEndpoint}"), credential: new DefaultAzureCredential());

            var keyOptions = new CreateKeyOptions();
            keyOptions.Enabled = true;
            keyOptions.KeyOperations.Add(KeyOperation.UnwrapKey);
            keyOptions.KeyOperations.Add(KeyOperation.WrapKey);

            var newKey = await kvClient.CreateKeyAsync(keyName, KeyType.Rsa, keyOptions);

            return newKey.Value.Properties.Version;
        }

        //Rotate RSA Key on Key Vault.
        public static async Task<string> RotateKeyAsync(string vaultEndpoint, string keyName)
        {
            var kvClient = new KeyClient(vaultUri: new Uri($"https://{vaultEndpoint}"), credential: new DefaultAzureCredential());

            var newKey = await kvClient.RotateKeyAsync(keyName);

            return newKey.Value.Properties.Version;
        }
    }
}